tests/test_well.py

```diff
@@ -61,3 +64,45 @@ def test_logs():
             discovered_logs += 1
 
    assert discovered_logs > 0
+
+
+# Deviation Survey tests
+
+def test_DeviationSurvey(example_model_with_well):
+
+   # Load existing objects, using type hints
+   model: Model
+   datum: resqpy.well.MdDatum
+   model, well_interp, datum, traj = example_model_with_well
+   epc_path = model.epc_file
+
+   # Create new survey
+   survey = resqpy.well.DeviationSurvey(
+      parent_model=model,
+      represented_interp=well_interp,
+      md_datum=datum,
+      measured_depths=[1,2,3],
```

Comment:
These four numerical arrays would normally be floating point numbers – testing with integers is a curiosity but perhaps testing with float data would be better.

Reply:
I updated the write_hdf functino to ensure the MDs are cast as floats when saved to HDF5.

Reply:
The functino sounds like something discovered at CERN!